### PR TITLE
side menu 2

### PR DESCRIPTION
**Pull request by benbaler**

### CODE REVIEW

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.ts [hunk 1]
```diff
@@ -37,12 +38,10 @@ export class SideMenuOptionComponent implements OnInit, OnChanges {
...
   }
 
-  ngOnInit() {
-    this.renderComponent();
-  }
-
-  ngOnChanges() {
-    this.renderComponent();
+  ngOnChanges(changes: SimpleChanges): void {
+    if (has(changes, 'options.prefix')) {
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.ts]: not changes.option? why plural it should get a single option

Comment (benbaler) [reply projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.ts]: done

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.scss [hunk 2]
```diff
@@ -1,34 +1,24 @@
...
-
-    &:hover {
-      background-color: #F9F9F9;
-    }
-
-    &.selected-option {
-      background-color: #F9F9F9;
-      font-weight: bold;
-    }
+    @include listOption;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.scss]: thanks

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.spec.ts [hunk 3]
```diff
@@ -52,7 +51,18 @@ describe('SideMenuOptionComponent', () => {
...
+          currentValue: getSideMenuOptionsMock()[0],
+          previousValue: {},
+          firstChange: true,
+          isFirstChange: () => true,
+        },
+      };
+      component.ngOnChanges(changes);
+      fixture.detectChanges();
+      prefixComponent = fixture.debugElement.query(By.css('.option-prefix b-icon'));
+      expect(prefixComponent).toBeTruthy();
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.spec.ts]: cool!

Comment (omrihecht) [reply projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.spec.ts]: You should probably add a mock for the icon service so you don't make requests to the http, angular tests do that
search for `spyIconService = createSpyObj('spyIconService', ['initIcon']);`

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.ts [hunk 4]
```diff
@@ -37,12 +38,10 @@ export class SideMenuOptionComponent implements OnInit, OnChanges {
...
   }
 
-  ngOnInit() {
-    this.renderComponent();
-  }
-
-  ngOnChanges() {
-    this.renderComponent();
+  ngOnChanges(changes: SimpleChanges): void {
+    if (has(changes, 'option.currentValue.prefix')) {
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.ts]: the condition is wrong
the interface of SimpleChanges is:
```
export interface SimpleChanges {
    [propName: string]: SimpleChange;
}
```

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.spec.ts (hunk 3 above)
Comment (omrihecht) [reply projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.spec.ts]: also might be good to add test for when `has(changes, 'option.currentValue.prefix')` is false, meaning test you are not rendering component

**Comment by benbaler** [comment]
@omrihecht there are tests for side-menu it was on the previous PR https://github.com/hibobio/bob-style/pull/290

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.html [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <div class="side-menu-option-container"
-     [ngClass]="{'selected-option': selected, 'no-prefix': !option.prefix}"
+     [ngClass]="{'selected': selected, 'no-prefix': !option.prefix}"
```
[the remark this answers is not in the document]
Comment (benbaler) [reply projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.html]: thanks

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.scss (hunk 2 above)
Comment (benbaler) [reply projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.scss]: 👍

**Pull request merged**